analysis/processor/processors/proc_source_code_load.go

```diff
@@ -0,0 +1,225 @@
+package processors
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/alecthomas/chroma/formatters/html"
+	"github.com/alecthomas/chroma/lexers"
+	"github.com/alecthomas/chroma/styles"
+	"github.com/deepsourcelabs/cli/types"
+)
+
+const sourceCodeOffset int = 3
+
+type ProcSourceCodeLoad struct{}
+
+type formattedFile struct {
+	fileContent        []string
+	highlightedContent []string
+}
+
+var (
+	lineStartWithOffset, lineEndWithOffset int
+	formattedFileCache                     map[string]formattedFile = make(map[string]formattedFile)
+)
+
+// We cache the past file's iterator, taking advantage of the fact that
+// issues are sorted according to filenames
+func getFinalFormattedSlice(fileContentSlice []string, issue *types.Issue) formattedFile {
+	filePath := issue.Location.Path
+
+	// Check if the formatted file data is already present in the cache.
+	// If yes, return the cached data.
+	if formattedFileData, ok := formattedFileCache[filePath]; ok {
+		return formattedFileData
+	}
+
+	// Else, clear the cache.
+	for k := range formattedFileCache {
+		delete(formattedFileCache, k)
+	}
+
+	/* ============================================================
+	 * Use alecthomas/chroma to generate syntax highlighted snippet
+	 * ============================================================ */
+	fileContentString := strings.Join(fileContentSlice, "\n")
+	lexer := lexers.Match(filePath)
+
+	/* Case: In case of .vue files, use the `html` lexer since the `vue` lexer
+	 * breaks in certain cases. The `html` lexer provides comparatively better results.
+	 * TODO(SNT): Remove this case if vue lexer is improved in future. */
+	if strings.HasSuffix(filePath, ".vue") {
+		lexer = lexers.Get("html")
+	}
+	if lexer == nil {
+		lexer = lexers.Fallback
+	}
+
+	// Tokenize the file content.
+	iterator, err := lexer.Tokenise(nil, fileContentString)
+	if err != nil {
+		log.Println("Could not tokenize file ", filePath)
+		return formattedFile{
+			fileContent:        []string{},
+			highlightedContent: []string{},
+		}
+	}
+
+	// Selecting the chroma format in which we expect the output(html) and use the `monokai` colorscheme to highlight the snippet.
+	formatter := html.New(html.WithLineNumbers(true), html.PreventSurroundingPre(true), html.WithClasses(true))
+	style := styles.Get("monokai")
+	if style == nil {
+		style = styles.Fallback
+	}
+
+	var chromaFormattedBytes bytes.Buffer
+	var chromaFormattedString string
+	err = formatter.Format(&chromaFormattedBytes, style, iterator)
+	if err != nil {
+		fmt.Println(err)
```

Comment:
```suggestion
```

Reply:
This is required to showcase any error that occurs while formatting a file.